elements/pfe-button/src/pfe-button.js

```diff
@@ -0,0 +1,192 @@
+
+  constructor() {
+    super(PfeButton, { type: PfeButton.PfeType });
+
+    this._init = this._init.bind(this);
+    this._parentObserverHandler = this._parentObserverHandler.bind(this);
+    this._clickHandler = this._clickHandler.bind(this);
+    this._internalBtnContainer = this.shadowRoot.querySelector("#internalBtn");
+    this._observer = new MutationObserver(this._parentObserverHandler);
+    this._externalBtnClickHandler = this._externalBtnClickHandler.bind(this);
+    this._externalBtnObserver = new MutationObserver(this._init);
+
+    this.addEventListener("click", this._clickHandler);
+  }
+
+  connectedCallback() {
+    super.connectedCallback();
+    this._externalBtn = this.querySelector("button");
+
+    if (this.children.length) {
+      this._init();
+    }
+
+    this._observer.observe(this, parentObserverConfig);
+
+    if (this._externalBtn) {
+      this._externalBtnObserver.observe(
+        this._externalBtn,
+        externalBtnObserverConfig
+      );
+    }
+  }
+
+  disconnectedCallback() {
+    this.removeEventListener("click", this._clickHandler);
+    this._observer.disconnect();
+    this._externalBtnObserver.disconnect();
+  }
+
+  attributeChangedCallback(attr, oldVal, newVal) {
+    super.attributeChangedCallback(attr, oldVal, newVal);
+
+    switch (attr) {
+      case "disabled":
+        if (!this._externalBtn) {
+          return;
+        }
+
+        if (this.disabled) {
+          this._externalBtn.setAttribute("disabled", "");
+        } else {
+          this._externalBtn.removeAttribute("disabled");
+        }
+        break;
+    }
+  }
+
+  _init() {
+    if (!this._isValidLightDom()) {
+      return;
+    }
+
+    if (!this._externalBtn) {
+      return;
+    }
+
+    this._externalBtnObserver.disconnect();
+
+    if (this._externalBtn.hasAttribute("disabled")) {
+      this.setAttribute("disabled", "");
+    } else {
+      this.removeAttribute("disabled");
+    }
+
+    const clone = this._externalBtn.cloneNode(true);
+    denylistAttributes.forEach(attribute => {
+      if (clone.hasAttribute) {
+        clone.removeAttribute(attribute);
+      }
+    });
+
+    this._internalBtnContainer.innerHTML = clone.outerHTML;
+    this._externalBtnObserver.observe(
+      this._externalBtn,
+      externalBtnObserverConfig
+    );
+
+    this._externalBtn.addEventListener("click", this._externalBtnClickHandler);
+  }
+
+  _isValidLightDom() {
+    if (!this.children.length) {
+      console.warn(`${PfeButton.tag}: You must have a button in the light DOM`);
+      return false;
+    }
+
+    if (this.children[0].tagName !== "BUTTON") {
+      console.warn(
+        `${PfeButton.tag}: The only child in the light DOM must be a button tag`
+      );
+
+      return false;
+    }
+
+    return true;
+  }
+
+  // when the parent changes, make sure the light DOM is valid,
+  // set the _externalBtn, and initialize the component
+  _parentObserverHandler() {
+    if (!this._isValidLightDom()) {
+      return;
```

Comment:
@kylebuch8 
Just a comment, what happens if it _was_ valid light DOM, but becomes invalid light DOM? Throw an error and maintain current state? Is that the right behavior in that circumstance, was thinking maybe it should get disabled?